Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/TransformToIqt.py

```diff
@@ -213,29 +225,45 @@ def _transform(self):
         Run TransformToIqt.
         """
         from IndirectCommon import CheckHistZero, CheckHistSame, CheckAnalysers
+
         try:
             CheckAnalysers(self._sample, self._resolution)
         except ValueError:
             # A genuine error the shows that the two runs are incompatible
             raise
-        except:
-            # Checking could not be performed due to incomplete or no instrument
-            logger.warning('Could not check for matching analyser and reflection')
+        except BaseException:
+            # Checking could not be performed due to incomplete or no
+            # instrument
+            logger.warning(
+                'Could not check for matching analyser and reflection')
 
         # Process resolution data
         num_res_hist = CheckHistZero(self._resolution)[0]
         if num_res_hist > 1:
-            CheckHistSame(self._sample, 'Sample', self._resolution, 'Resolution')
-
-        iqt = CalculateIqt(InputWorkspace=self._sample, ResolutionWorkspace=self._resolution, EnergyMin=self._e_min,
-                           EnergyMax=self._e_max, EnergyWidth=self._e_width,
-                           NumberOfIterations=self._number_of_iterations, SeedValue=self._seed,
-                           StoreInADS=False, OutputWorkspace="__ciqt")
+            CheckHistSame(
+                self._sample,
+                'Sample',
+                self._resolution,
+                'Resolution')
+
+        calculateiqt_alg = self.createChildAlgorithm(name='CalculateIqt', startProgress=0.3,
+                                                     endProgress=1.0, enableLogging=True)
+        calculateiqt_alg.setAlwaysStoreInADS(False)
+        args = {"InputWorkspace": self._sample, "OutputWorkspace": "iqt", "ResolutionWorkspace": self._resolution,
+                "EnergyMin": self._e_min, "EnergyMax": self._e_max, "EnergyWidth": self._e_width,
+                "CalculateErrors": self._calculate_errors, "NumberOfIterations": self._number_of_iterations,
+                "SeedValue": self._seed}
+        for key, value in args.items():
```

Comment:
As a side note - `args.items()` (Python 2.X) returns a copy of the items, whereas using `from six import iteritems` and `iteritems(args)` returns an iterator (lower memory usage). This is a small dict so shouldn't make too much difference, but worth remembering.